src/libpmemfile-posix/write.c

```diff
@@ -202,6 +202,8 @@ pmemfile_pwritev_internal(PMEMfilepool *pfp,
 	os_rwlock_unlock(&vinode->rwlock);
 
 	if (error) {
+		if (error == ENOMEM)
+			error = ENOSPC;
```

Comment:
Please move this to TX_ONABORT section. When vinode_rebuild_block_tree fails with ENOMEM I want to get ENOMEM instead of ENOSPC. (Eventually this will be pushed down to pmemobj_tx_xalloc call site once we'll get NO_ABORT_ON_FAIL flag)